### ISSUE TITLE
[Question] Anybody building python extensions using clang++?

### ISSUE THREAD

**Issue by lanctot** [joerick/cibuildwheel#638]
I have not been able to find clang++ in the standard path on the manylinux images, just wondering if they are missing it or if I have something badly setup.

Wheels config file here: https://github.com/lanctot/open_spiel/blob/master/.github/workflows/wheels.yml

Our project uses pybind11 and requires C++17. Thanks!

**Comment by Czaki** [comment]
As far as I know, any of the default manylinux images does not have clang installed by default. 

Did you need something specific dependencies to clang or only need to have c++17 support? 

g++ also supports c++17 https://gcc.gnu.org/projects/cxx-status.html#cxx17 most of them are supported from version 7. 

manylinux2014 should contain enough new version of gcc. Maybe even manylinux2010 even could build it.

Could you check if using the modern version of gcc is enough?

**Comment by YannickJadoul** [comment]
`cibuidlwheel` is not in control over what's in the `manylinux` images. If it's available, `yum install` (or `apt install` in the newest ones) should work, though?

**Comment by lanctot** [comment]
Thank you for the replies. 

Our project stopped (officially) supporting building from g++ last October because we did not want the maintenance burden of handling different compilers, and we all use clang++. In retrospect, that was maybe unnecessary, and msy need to be reconsidered :)

I tried now and seems to compile and run tests fine under g++, and I want to use the default tools in the docker images so I will just use g++ to build the wheels and make sure they both work.

Thanks again, and wonderful project. Will make in distributing our code much easier!

**Comment by henryiii** [comment]
I would usually recommend supporting (at least) the default toolchains on OS's. That's GCC on Linux, Clang on macOS, and MSVC on Windows. Supporting multiple toolchains (on different OSs, admittedly) often means you have better, standards-compliant code and different compilers can catch different bugs. The standard toolchains are often assumed at least in Python packaging, like in building wheels, or even on conda-forge, where you are expected to build with the default toolchain.

**Comment by Czaki** [comment]
Also clang on macos and linux is not the same tool. For example clang on macos does not support openmp and it is not so simple to use it without homebrew (which limits supported version of systems).

**Comment by lanctot** [comment]
Thanks everyone for all the help and advice. I have now switched to using g++ when building wheels on Linux and clang++ when building wheels on MacOS.

I'm now running into an issue where CMake cannot find the Python dev headers, which is quite surprising to me given how simple all the examples seem, so maybe I don't have something setup correctly. Anyway I've opened another issue if anybody is curious: https://github.com/joerick/cibuildwheel/issues/639
